### PR TITLE
[BEAM-13906] Improve coverage of errors package

### PR DESCRIPTION
**Pull request by damccorm**
Improve test coverage of our errors package. Mostly, this was covering a few more nil edge cases and the beamError.Error() function

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damccorm** [comment]
R: @jrmccluskey @lostluck

**Review by jrmccluskey — APPROVED** [review]
LGTM! Looks like the gaps in testing here were largely the nil routes, which seems to be the theme for a lot of the under-tested packages. Good job!

File: sdks/go/pkg/beam/internal/errors/errors_test.go [hunk 1]
```diff
@@ -147,10 +170,51 @@ func TestSetTopLevelMsgf(t *testing.T) {
...
+		},
+	}
+
+	for _, test := range tests {
+		if be, ok := test.err.(*beamError); ok {
+			if got, want := be.Error(), test.want; got != want {
+				t.Errorf("Incorrect formatting. Want: %q, Got: %q", want, got)
+			}
+		} else {
+			t.Errorf("Error should be type beamError, got: %q", test.err)
```
Comment (lostluck) [inline sdks/go/pkg/beam/internal/errors/errors_test.go:206]: Types and their pointers are distinct types, even though the pointer variant is derived. I'm going to commit the suggested edit but wanted to clarify what it was I was doing.

```suggestion
			t.Errorf("Error should be type *beamError, got: %q", test.err)
```

**Review by lostluck — APPROVED** [review]
LGTM

**Pull request merged**